ssl/statem/statem_srvr.c

```diff
@@ -1781,6 +1781,16 @@ int tls_construct_server_key_exchange(SSL *s)
             n += 1 + nr[i];
         else
 #endif
+#ifndef OPENSSL_NO_DH
+        /*
+         * for interoperability with some versions of the Microsoft TLS
+         * stack, we need to zero pad the DHE pub key to the same length
+         * as the prime, so include the length of the prime here
+         */
+        if ((i == 2) && (type & (SSL_kDHE | SSL_kDHEPSK)) && (nr[0] > nr[2]))
```

Comment:
Personally I think it is clearer without the expression. The comment above explains why we're using the length of the prime rather than the length of pub key...so lets just always use the length of the prime, i.e. "use the length of the prime" is a simpler thing to understand than "use the length of the prime unless the length of the pubkey equals the length of the prime in which case use the length of the pubkey". Similarly with the other instance of this.